api/controller/sshtunneler/client.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshtunneler
+
+import (
+	"github.com/juju/names/v5"
+
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/state"
+)
+
+// Client holds the SSH server client for it's respective worker.
+type Client struct {
+	facade base.FacadeCaller
+}
+
+// NewClient returns an SSH server facade client.
+func NewClient(caller base.APICaller) (*Client, error) {
+	facadeCaller := base.NewFacadeCaller(caller, "SSHTunneler")
+	return &Client{
+		facade: facadeCaller,
+	}, nil
+}
+
+// ControllerAddresses returns a list of controller addresses for the specified machine.
```

Comment:
public addresses

Reply:
Technically the function executing on the facade side, `m.Addresses().AllMatchingScope(network.ScopeMatchPublic).Values()...)` returns all addresses in an order that matches the scope `ScopeMatchPublic`, then `ScopeCloudLocal` then `ScopeFanLocal, ScopeUnknown`. So it's a bit more than purely just the public addresses.

Reply:
Updated the docstring to "potentially reachable controller addresses"